### PR TITLE
Preserve old conda packages when new fail to generate

### PR DESCRIPTION
**Pull request by guyang3532**

### CODE REVIEW

File: prune.sh [hunk 1]
```diff
@@ -9,6 +9,24 @@ grab_version() {
...
+grab_latest_version() {
+    conda search -c "${CHANNEL}" --platform "${platform}" "${PKG}" 2>/dev/null | \
+        grep "${CHANNEL}" | \
+        awk -F '  *' '{print $2}' | \
+        uniq | \
+        tail -n 1 | \
+        xargs
+}
+
+grab_builds() {
```
Comment (peterjc123) [inline prune.sh]: nit: add a comment here (it returns the spec of the package of a specific version `$1`)

File: prune.sh [hunk 2]
```diff
@@ -17,21 +35,19 @@ CHANNEL=${CHANNEL:-pytorch-nightly}
...
-    unique_versions["${version}"]=""
-done
-
-for version in ${!unique_versions[@]}; do
-    (
-        set -x
-        anaconda remove --force ${CHANNEL}/${PKG}/${version}
-    )
-done
+    latest_version="$(grab_latest_version || true)"
```
Comment (peterjc123) [inline prune.sh:39]: nit: formatting

File: prune.sh [hunk 3]
```diff
@@ -17,21 +35,19 @@ CHANNEL=${CHANNEL:-pytorch-nightly}
...
+	latest_builds="$(grab_builds "${latest_version}" || true)"
+    versions_to_prune="$(grab_version || true)"
+	for version in ${versions_to_prune}; do
+		builds="$(grab_builds "${version}" || true)"
+		for build in ${builds}; do
+		    if [[ "${latest_builds}" =~ "${build}" ]];then
+			    (
+			        set -x
+			        anaconda remove --force ${CHANNEL}/${PKG}/${version}/${PLATFORMS}/${PKG}-${version}-${build}.tar.bz2
+				)
```
Comment (peterjc123) [inline prune.sh]: nit: formatting

File: prune.sh [hunk 4]
```diff
@@ -17,21 +35,19 @@ CHANNEL=${CHANNEL:-pytorch-nightly}
...
+			fi
+        done
+    done
+done
```
Comment (peterjc123) [inline prune.sh]: nit: formatting

File: prune.sh (hunk 2 above)
Comment (guyang3532) [reply prune.sh:39]: > nit: formatting

done

**Review by peterjc123 — APPROVED** [review]

File: prune.sh (hunk 1 above)
Comment (guyang3532) [reply prune.sh]: > nit: add a comment here (it returns the spec of the package of a specific version `$1`)

yes, I modify the function name.

**Review by seemethere — APPROVED** [review]

**Pull request merged**